src/bindings/python/tests/test_runtime/test_infer_request.py

```diff
@@ -97,7 +98,12 @@ def abs_model_with_data(device, ov_type, numpy_dtype):
 
 def test_get_profiling_info(device):
     core = Core()
-    param = ops.parameter([1, 3, 32, 32], np.float32, name="data")
+    if platform.system() == "Darwin" and platform.machine() == "arm64":
+        # arm64 prefers fp16, and fp32 input will trigger a convert node
+        # to be added, so assert 'Softmax' will failed.
+        param = ops.parameter([1, 3, 32, 32], np.float16, name="data")
+    else:
+        param = ops.parameter([1, 3, 32, 32], np.float32, name="data")
```

Comment:
@akuporos, could you please take a look at the test change?

Reply:
API 2.0 have new precision logic than legacy API, which behavior working with  https://github.com/openvinotoolkit/openvino/pull/19993 will bring such issue.

Reply:
I am not sure about `"Darwin"` being checked, is it really system related?

Also I think that extending to `platform.machine() in ["arm", "armv7l", "aarch64", "arm64", "ARM64"]` is necessary to catch all possible arm-related names.
Otherwise you may try to get property called `FULL_DEVICE_NAME` and look for `arm_compute::NEON` in it. However, I am not sure if it will work with API 2.0, it was for sure valid with 1.0.

Reply:
Shouldn't we check that the `device` is actually a CPU?

Reply:
@maxnick is there a scenario where ARM-based check run other devices? If yes, the check may include `os.environ.get("TEST_DEVICE", "CPU") == "CPU"` assuming that `CPU` is used by default.

Reply:
I have referred to below code to check platform/device, and "Darwin" should be MacOS M1:
https://github.com/openvinotoolkit/openvino/blob/5cd9659033ef37d9f61203eb4630ee3add54ed01/tests/layer_tests/pytorch_tests/test_nms.py#L40

There is a platform checking code for python wheel:
https://github.com/openvinotoolkit/openvino/blob/5cd9659033ef37d9f61203eb4630ee3add54ed01/src/bindings/python/wheel/setup.py#L32-L40

Reply:
How about:
`if platform.machine() == "arm64" and device == "CPU":`

Reply:
I think we discuss not essential thing. At first glance it's not clear why we should modify the parameter precision. Even though the plugin inserts a Convert node, shouldn't break the fact that the SoftMax node still persists. May be the search logic in the test should be fixed?